libfwupdplugin/fu-device.h

```diff
@@ -193,6 +193,35 @@ FuDevice	*fu_device_new				(void);
 #define fu_device_get_flashes_left(d)		fwupd_device_get_flashes_left(FWUPD_DEVICE(d))
 #define fu_device_get_install_duration(d)	fwupd_device_get_install_duration(FWUPD_DEVICE(d))
 
+/**
+ * FuDeviceFlags:
+ * @FU_DEVICE_FLAG_NONE:			No flags set
+ * @FU_DEVICE_FLAG_NO_AUTO_INSTANCE_IDS:	Do not add instance IDs from the device baseclass
+ * @FU_DEVICE_FLAG_ENSURE_SEMVER:		Ensure the version is a valid semantic version, e.g. numbers separated with dots
+ * @FU_DEVICE_FLAG_ONLY_SUPPORTED:		Only devices supported in the metadata will be opened
+ * @FU_DEVICE_FLAG_MD_SET_NAME:			Set the device name from the metadata `name` if available
+ * @FU_DEVICE_FLAG_MD_SET_NAME_CATEGORY:	Set the device name from the metadata `category` if available
+ * @FU_DEVICE_FLAG_MD_SET_VERFMT:		Set the device version format from the metadata if available
+ * @FU_DEVICE_FLAG_MD_SET_ICON:			Set the device icon from the metadata if available
+ *
+ * The device internal flags.
+ **/
+typedef enum {
+	FU_DEVICE_FLAG_NONE			= 0,
+	FU_DEVICE_FLAG_NO_AUTO_INSTANCE_IDS	= (1llu << 19),	/* Since: 1.5.5 */
```

Comment:
Presumably you don't want to start at 19..

Reply:
That was actually deliberate, if a plugin author accidentally used the deprecated #define rather than the new one then it all magically just works. The other way would be to check if any of the new FU_DEVICE_FLAGS are >= `1 << 19` for a few months and critical warn if so so that plugin authors know they messed up.

Reply:
We could easily have a CI script that greps for stuff like `FWUPD_DEVICE_FLAG_MD_SET_ICON`  in the diff until ABI bump so we don't end up losing 18 more of these by "starting" at 19.

Reply:
> We could easily have a CI script that greps for stuff like FWUPD_DEVICE_FLAG_MD_SET_ICON in the diff

This is an excellent idea. I've added this and renumbered the internal flags to start at 0 as expected.